code-journal/js/main.js

```diff
@@ -134,15 +145,28 @@ function viewSwapping(dataView) {
 
 // listen for DOMContentLoaded
 
-
 document.addEventListener('DOMContentLoaded', function (event) {
-  // var dataFromLocal = localStorage.getItem('javascript-local-storage')
-  // data = JSON.parse(dataFromLocal);
-
   if (data.profile.username === undefined) {
     viewSwapping('edit-profile');
   } else {
     viewSwapping('profile');
   }
 
-})
+});
+
+// listen for link
+
+document.addEventListener('click', function (event) {
+  var $dataview = event.target.getAttribute('data-view');
+  if (event.target.tagName !== 'A') {
+    return;
+  }
+  if (data.profile.username === '') {
+    return;
+  }
+  if ($dataview !== 'edit-profile') {
```

Comment:
With this conditional, the only valid view on load is edit profile, which doesn't make much sense if the user already has profile info. This should be an `else` statement for the above `if` statement after you revise it. If they have a username, that means they have a profile, so they can navigate to whichever view they requested :grin: